pages/scan.tsx

```diff
@@ -198,10 +202,13 @@ Scan.getInitialProps = async ctx => {
   const profile = await getProfile(req);
 
   // Null profile means user is not logged in, and this is only relevant for admins
-  if (!profile || profile.role !== "admin") {
+  if (!profile || !(profile.role == "admin" || profile.role == "volunteer" || profile.role == "sponsor")) {
     handleLoginRedirect(req);
   }
 
+  // jenk but fix later
+  req.user.role = profile.role
```

Comment:
Can we assume the `req.user.role` to be defined everywhere in the application? not sure where this is getting set besides here

Reply:
no, that's why there's a null check in tasks.js

Reply:
Kk